pool/pool_aio_test.go

```diff
@@ -173,35 +347,266 @@ func TestPoolInterfaceWithAIO(t *testing.T) {
+	t.Run("upload object", func(t *testing.T) {
+		var prmSess client.PrmSessionCreate
+		prmSess.SetExp(math.MaxUint64)
+		prmSess.UseSigner(signer)
+
+		res, err := cl.SessionCreate(ctx, prmSess)
+		require.NoError(t, err)
+
+		var id uuid.UUID
+		err = id.UnmarshalBinary(res.ID())
+		require.NoError(t, err)
+
+		var key neofsecdsa.PublicKey
+		err = key.Decode(res.PublicKey())
+		require.NoError(t, err)
+
+		var sess session.Object
+
+		sess.SetID(id)
+		sess.SetAuthKey(&key)
+		sess.SetExp(math.MaxUint64)
+		sess.ForVerb(session.VerbObjectPut)
+		sess.BindContainer(containerID)
+
+		err = sess.Sign(signer)
 		require.NoError(t, err)
 
-		ctxTimeout, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+		ctxTimeout, cancel := context.WithTimeout(ctx, defaultTimeOut)
 		defer cancel()
 
-		err = isObjectDeleted(ctxTimeout, pool, containerID, objectID)
+		rf := object.RequiredFields{
+			Container: containerID,
+			Owner:     account,
+		}
+
+		var hdr object.Object
+		object.InitCreation(&hdr, rf)
+
+		var prm client.PrmObjectPutInit
+		prm.UseSigner(signer)
+		prm.SetCopiesNumber(1)
+		prm.WithinSession(sess)
+
+		w, err := cl.ObjectPutInit(ctxTimeout, hdr, prm)
+		require.NoError(t, err)
+
+		require.True(t, w.WritePayloadChunk(payload))
+
+		resp, err := w.Close()
 		require.NoError(t, err)
+
+		objectID = resp.StoredObjectID()
+	})
+
+	t.Run("download object", func(t *testing.T) {
+		ctxTimeout, cancel := context.WithTimeout(ctx, defaultTimeOut)
+		defer cancel()
+
+		var cmd client.PrmObjectGet
+		cmd.UseSigner(signer)
+
+		hdr, read, err := cl.ObjectGetInit(ctxTimeout, containerID, objectID, cmd)
+		defer func() {
+			_ = read.Close()
+		}()
+
+		require.NoError(t, err)
+		require.NotNil(t, hdr.OwnerID())
+		require.True(t, hdr.OwnerID().Equals(account))
+
+		downloadedPayload := make([]byte, len(payload))
+
+		l, ok := read.ReadChunk(downloadedPayload)
+		require.True(t, ok)
+		require.Equal(t, l, len(payload))
+
+		require.True(t, bytes.Equal(payload, downloadedPayload))
+	})
+
+	t.Run("delete object", func(t *testing.T) {
+		ctxTimeout, cancel := context.WithTimeout(ctx, defaultTimeOut)
+		defer cancel()
+
+		testDeleteObject(t, ctxTimeout, signer, containerID, objectID, cl)
 	})
 
 	t.Run("container delete", func(t *testing.T) {
-		var cmd client.PrmContainerDelete
-		cmd.SetSigner(signer)
+		ctxTimeout, cancel := context.WithTimeout(ctx, defaultTimeOut)
+		defer cancel()
 
-		require.NoError(t, pool.ContainerDelete(ctx, containerID, cmd))
+		testDeleteContainer(t, ctxTimeout, signer, containerID, wait)
+	})
 
-		ctxTimeout, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+	t.Run("container really deleted", func(t *testing.T) {
+		ctxTimeout, cancel := context.WithTimeout(ctx, defaultTimeOut)
 		defer cancel()
 
-		err = isBucketDeleted(ctxTimeout, pool, containerID)
 		require.NoError(t, err)
+		require.NoError(t, isBucketDeleted(ctxTimeout, cl, containerID))
```

Comment:
I'd expect it to be a simple `get` of a container. Suppose I've removed polling from `waiter`, this test will still succeed because it retries/waits internally.